File: HLIN504/fact.c
#include <stdlib.h>
#include <stdio.h>

double fact(double n){
	double resultat = 1;
	for (int i = 1; i <= n; i++){
		resultat *= i;
	}
	return resultat;
}

int main()
{
	double valeur;
	printf("Entrer une valeur à calculer :\n");
	scanf("%lf", &valeur);
	printf("Le resultat est : %lf\n", fact(valeur));
	return 0;
}

File: HLIN510/fact.php
<!DOCTYPE html>
<html>
<head>
    <title></title>
</head>
<body>

    <form name="F" method="get" action='<?php echo $_SERVER['PHP_SELF']; ?>'>
        <input type="number" name="valeur">
        <input type="submit" name="ok" value="Calculer">
    </form>

    <?php
    function fact($nbr) 
    { 
       if($nbr === 0) // condition d'arret 
       return 1;  
       else 
        return $nbr*fact($nbr-1); 
} 
if (isset($_GET['ok'])){
    echo "Resultat: ", $_GET['valeur'], "! = ", fact($_GET['valeur']);
}
?>
</body>
</html>